### PR TITLE
Small fixes for Jenkins to CloudFoundry deployment

### PR DESCRIPTION
**Pull request by r4vi**
* make Directory sso env vars optional
* move Procfile to top level
* add whitenoise to wsgi.conf

### CODE REVIEW

File: conf/gconfig.py [hunk 1]
```diff
@@ -1,4 +1,7 @@
-bind = "0.0.0.0:8080"
+import os
+
+PORT = os.environ.get("PORT", 8080)
+bind = "0.0.0.0:{PORT}".format(PORT=PORT)
```
Comment (saruniitr) [inline conf/gconfig.py:4]: minor, may be replace with `f"0.0.0.0:{PORT}"`

File: Procfile [hunk 2]
```diff
@@ -0,0 +1 @@
+web: ./manage.py migrate && ./manage.py compilescss && ./manage.py collectstatic --ignore=*.scss,*.md,*.txt,*.json,LICENSE,license,CHANGES,changes && gunicorn --preload -c conf/gconfig.py -b 0.0.0.0:$PORT conf.wsgi
```
Comment (kerin) [inline Procfile:1]: Are we leaving sorting this out for a future PR?

Comment (r4vi) [reply Procfile:1]: yes, just want to get it working now

File: conf/gconfig.py (hunk 1 above)
Comment (r4vi) [reply conf/gconfig.py:4]: not sure we can use f-strings as I'm not sure if the project is python 3.6 or 3.7

Comment (kerin) [reply conf/gconfig.py:4]: `runtime.txt` sez `python-3.6.9`

Comment (r4vi) [reply conf/gconfig.py:4]: pipfile says 3.7 but either way I should've used f-strings since they came in with 3.6 :disappointed:

Comment (kerin) [reply conf/gconfig.py:4]: oh yes, having the same python version specified in runtime.txt and Pipfile is on the tech debt list - worth trying bumping everything to 3.7.x latest?

**Pull request merged**